src/basic/namespace-util.c

```diff
@@ -471,17 +470,12 @@ int userns_acquire_empty(void) {
                 /* Child. We do nothing here, just freeze until somebody kills us. */
                 freeze();
 
-        r = namespace_open(pid, NULL, NULL, NULL, &userns_fd, NULL);
-        if (r < 0)
-                return log_error_errno(r, "Failed to open userns fd: %m");
-
-        return TAKE_FD(userns_fd);
+        return pidref_namespace_open_by_type(&PIDREF_MAKE_FROM_PID(pid), NAMESPACE_USER);
```

Comment:
i wonder if PIDREF_MAKE_FROM_PID() really is the way to go, because it means pidref_namespace_open_by_type_internal() will become dependent on /proc/ for no reason. maybe switch to pidref_safe_fork() here, so that /proc/ is no necessary, at least on current kernels.

(here and below)